ts/navigation/TabNavigator.tsx

```diff
@@ -200,21 +200,23 @@ export const MainTabNavigator = () => {
             )
           }}
         />
-        <Tab.Screen
-          name={ROUTES.PROFILE_MAIN}
-          component={ProfileMainScreen}
-          options={{
-            title: I18n.t("global.navigator.profile"),
-            tabBarIcon: ({ color, focused }) => (
-              <TabIconComponent
-                iconName="navProfile"
-                iconNameFocused="navProfileFocused"
-                color={color}
-                focused={focused}
-              />
-            )
-          }}
-        />
+        {!showBarcodeScanSection && (
```

Comment:
The profile icon should be hidden only if we have 5+ items in the navbar, so if have the new wallet section enabled (which splits the wallet into two separate tabs) and the barcode scan button together.
I think this check should also include `isNewWalletSectionEnabled`

Reply:
Thanks for the suggestion! I've add `isSettingsVisibleAndHideProfileSelector` to implement this logic!